tests/test_profiling_utils.py

```diff
@@ -11,17 +11,19 @@
 from unittest.mock import patch
 
 import pytest
+from torch import cuda
```

Comment:
Torch is a soft dependency so it shouldn't error if Torch isn't installed, it should just skip tests.

Reply:
There are other tests that require but do not verify torch installation (test_data_extraction.py, for example).  Not sure what you want to do there.

Reply:
Yeah, @jturner65 been meaning to fix that.

Reply:
Just use a try: except ImportError boolean

Reply:
pytest.importorskip seems a bit more elegant, no?